src/main/scala/ru/d4team/chat/dao/PersonDAO.scala

```diff
@@ -0,0 +1,36 @@
+package ru.d4team.chat.dao
+
+import io.getquill._
+import io.getquill.jdbczio.Quill
+import ru.d4team.chat.models.person.{Person, PersonRecord}
+import zio.{Task, URLayer, ZLayer}
+
+object PersonDAO {
+  type Env = Quill.Postgres[SnakeCase]
+  val live: URLayer[Env, PersonDAO] = ZLayer.fromFunction(PersonDAOImpl.apply _)
+}
+
+trait PersonDAO {
+  def addPerson(person: Person): Task[Person]
+  def findPerson(name: String): Task[Option[Person]]
+  def getPersons: Task[List[Person]]
+}
+
+final case class PersonDAOImpl(quill: Quill.Postgres[SnakeCase]) extends PersonDAO {
+
+  import quill._
+
+  private val schema = dynamicQuerySchema[PersonRecord]("messaging.person")
+
+  override def addPerson(person: Person): Task[Person] = run {
+    schema.insertValue(PersonRecord.fromPerson(person))
+  }.as(person)
+
+  override def findPerson(name: String): Task[Option[Person]] = run {
+    schema.filter(_.name == lift(name)).take(1)
+  }.map(_.headOption.map(Person.fromRecord))
+
+  override def getPersons: Task[List[Person]] = run {
+    schema
+  }.map(_.map(Person.fromRecord))
```

Comment:
As an option, we can add additional method for list transformation:
```scala
def fromRecords(rs: List[PersonRecord]): List[Person] = rs.map(fromRecord)
```
And then simplify method:
```scala
run(schema).map(Person.fromRecords)
```